api/v1alpha1/oadp_types.go

```diff
@@ -406,8 +406,10 @@ func (dpa *DataProtectionApplication) AutoCorrect() {
 		if pvOperationTimeout, err := time.ParseDuration(resticTimeout); err == nil && dpa.Spec.Configuration.Velero.Args.PodVolumeOperationTimeout == nil {
 			dpa.Spec.Configuration.Velero.Args.PodVolumeOperationTimeout = &pvOperationTimeout
 		}
-	}
-	if dpa.Spec.Configuration.Velero.Args != nil && dpa.Spec.Configuration.Velero.Args.RestoreResourcePriorities == "" {
+		if dpa.Spec.Configuration.Velero.Args.RestoreResourcePriorities == "" {
+			dpa.Spec.Configuration.Velero.Args.RestoreResourcePriorities = "securitycontextconstraints,customresourcedefinitions,namespaces,storageclasses,volumesnapshotclass.snapshot.storage.k8s.io,volumesnapshotcontents.snapshot.storage.k8s.io,volumesnapshots.snapshot.storage.k8s.io,datauploads.velero.io,persistentvolumes,persistentvolumeclaims,serviceaccounts,secrets,configmaps,limitranges,pods,replicasets.apps,clusterclasses.cluster.x-k8s.io,endpoints,services,-,clusterbootstraps.run.tanzu.vmware.com,clusters.cluster.x-k8s.io,clusterresourcesets.addons.cluster.x-k8s.io"
+		}
+	} else {
```

Comment:
That's what you wanted @kaovilai ?

Reply:
So if args is nil, then the next line would result in a nil pointer crash since it would result in calling `nil.RestoreResourcePriorities`. Instead of this, above line 399, if `Args` is nil, then set it to a non-nil value. Then the rest of the logic should work.

Reply:
right, going to work on the fix, thanks for the review

Reply:
Applied fix to nil pointer. I am still not very familiar with Go, if you have a better approach in mind, please suggest it

Reply:
Yeah see my comment above, if you need help with the example let me know

Reply:
ignore my comment I can't review code :)